kernel-rpm-script

```diff
@@ -0,0 +1,302 @@
+#!/bin/sh
+
+op=$(echo "$0" | sed -e "s/.*-//")
+
+while true ; do
+    case $1 in
+	--version)
+	    version="$2"
+	    shift 2
+	    ;;
+	--release)
+	    release="$2"
+	    shift 2
+	    ;;
+	--name)
+	    name="$2"
+	    shift 2
+	    ;;
+
+	--kernelrelease)
+	    kernelrelease="$2"
+	    shift 2
+	    ;;
+	--flavor)
+	    flavor="$2"
+	    shift 2
+	    ;;
+	--variant)
+	    flavor="$2"
+	    shift 2
+	    ;;
+
+	--usrmerged)
+	    usrmerged="$2"
+	    shift 2
+	    ;;
+	--image)
+	    image="$2"
+	    shift 2
+	    ;;
+	--systemmap)
+	    systemmap="$2"
+	    shift 2
+	    ;;
+
+	--modulesdir)
+	    modulesdir="$2"
+	    shift 2
+	    ;;
+	--basepackage)
+	    basepackage="$2"
+	    shift 2
+	    ;;
+	--certs)
+	    certs="$2"
+	    shift 2
+	    ;;
+	*) break
+	    ;;
+    esac
+done
+
+wm2=/usr/lib/module-init-tools/weak-modules2
+nvr="$name"-"$version"-"$release"
+
+trigger_purge_kernels() {
+    [ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || echo Triggering purge-kernels > /dev/kmsg
+    touch /boot/do_purge_kernels
+}
+
+run_wm2() {
+    [ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || echo wm2 "$@" > /dev/kmsg
+    /bin/bash -${-/e/} $wm2 "$@"
+}
+
+message_install_bl () {
+    echo "You may need to setup and install the boot loader using the"
+    echo "available bootloader for your platform (e.g. grub, lilo, zipl, ...)."
+}
+
+run_bootloader () {
+    if [ -f /etc/sysconfig/bootloader ] &&
+	[ -f /boot/grub/menu.lst -o \
+	-f /etc/lilo.conf      -o \
+	-f /etc/elilo.conf     -o \
+	-f /etc/zipl.conf      -o \
+	-f /etc/default/grub    ]
+	then
+	    return 0
+	else
+	    return 1
+    fi
+}
+
+[ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || \
+    echo "$op" name: "$name" version: "$version" release: "$release" \
+    kernelrelease: "$kernelrelease" flavor: "$flavor" variant: "$variant" \
+    usrmerged: "$usrmerged" image: "$image" systemmap: "$systemmap" \
+    modulesdir: "$modulesdir" basepackage: "$basepackage" certs: "$certs" > /dev/kmsg
+
+script_rc=0
+
+case $op in
```

Comment:
This case statement is far too long. Please write shell functions and make the case statement just call them.

Reply:
ugh, these are the scripts from the kernel

I expect that much of this functionality will be restructured in the future or abolished entirely but at first I want to be as close to 1:1 as is reasonably possible.